deploy/infra/lib/helpers/base-lambda.ts

```diff
@@ -0,0 +1,28 @@
+import * as cdk from "aws-cdk-lib";
+import * as lambda from "aws-cdk-lib/aws-lambda";
+import { PolicyStatement } from "aws-cdk-lib/aws-iam";
+
+export interface BaseLambdaFunProps {
+  functionProps: lambda.FunctionProps;
+  vpcConfig: any;
+}
```

Comment:
nit and just curious, could this be an extend? 

```suggestion
export interface BaseLambdaFunProps extends lambda.FunctionProps{
  vpcConfig: any;
}
```

but does not matter at all, looks great